sfcasts/ep1/computed.md

```diff
@@ -1,228 +1,192 @@
-`<hr />`, basically everything except for the button, when we're in a collapsed
-state.
+Look at `sidebar.vue`. In the template, we want to hide the `<h5>`, the `<ul>` and the
+`<hr />`: basically everything except for the button.
 
 In `Vue`, there are two ways to do this and both of them come in the form of
-*directives*. At the end of this chapter we will have seen almost all of the important
-directives in `Vue`. You will be a Directive Master!
+*directives*.
 
 To make this work, I'm going to add a new `<div>` to wrap all the elements that we
-need hide and *then* apply a directive to it so that it will hide or show this element
-and all of its contents.
+need hide.
 
-The first way of doing this is via the `v-if` directive. We can write `v-if=""`, and
-inside of it, use a JavaScript expression. We'll say `!collapsed` so as to show this
-if we are *not* in the collapsed state. If we go over to our browser we don't even
-need to refresh because we have the hot module replacement on... and it works!
+The first directive is `v-if`. We can write `v-if=""` and,
+inside, use a JavaScript expression. We'll say `!collapsed`. So: show this if we
+are *not* collapsed. If we go over to our browser, we don't even
+need to refresh because we're using hot module replacement. And... it works!
 Much better!
 
 ## inspecting v-if and the v-show directive
 
-The key thing here, if you inspect the element on that button, is that with `v-if`
-the contents are removed entirely. So if we click on the button again, we can see the
-contents shown, and if we hide it, they are gone! They are effectively added and
-removed from the DOM as the condition in the directive evaluates to `true` or `false`.
-That is *great*!
-
-The other way to hide things is by using `v-show`. With `v-show` things are going to
-look the same, but the difference is that when we click in `>> collapse` the content
-is always there. `v-show` just adds a style rule to the top element to set `display`
-to `none` when it needs to. *Sneaky*!
- 
-So `v-show` and `v-if` really accomplish the same task, they just do it in
-fundamentally different ways. So which one is more appropriate? It just depends on the
-situation. `v-show` is better when you're hiding and showing things a lot. When hiding
-and showing is really fast. If we have something that really won't be changing a lot,
-`v-if` is the way to go, because, if you use `v-if` here, for example, you won't be
-spending any time even rendering the sidebar until you actually need it. This is
-especially useful if you have several different layers of components here, where it
-can make your whole application load faster instead of loading all of these things
-just to hide them next.
+The key thing here, if you inspect the HTML, is that with `v-if`
+the contents are removed *entirely*. So if we click the button again, the contents
+come back, and if we collapse, they're gone!
```

Comment:
If you compare to the paragraph above, it was all about saying less when possible

```diff
- The key thing here, if you inspect the element on that button,
+ The key thing here, if you inspect the HTML,
```

I think (but it's subjective) that:

> They are effectively added and
> removed from the DOM as the condition in the directive evaluates to `true` or `false`.

is self-evident, so I removed it. I didn't check the original script, but *I* may have even said this! I often say too much when recording initially.

Reply:
So my addition here has to do with the fact that you can put almost *anything* there and JavaScript will only care about how it evaluates. If it's `truthy` or `falsey`, so you can say `""` empty string is `falsey`, or `0` is `falsey`. Or you can pass an array: `[]` and it will be `truthy`, even if empty, and lots of other things as well. 

This is probably one place in the script where I might argue and say: let's keep is a bit more complicated so that we can relay this exact concept in a few words: "what it evaluates to".

Up to you :)

Reply:
Yep, it's subjective. I leaned towards removing it because we *do* eventually set one of these to an expression (well, a very simple expression - a variable).